packages/codec/lib/index.ts

```diff
@@ -16,7 +16,147 @@ require("source-map-support/register");
 import * as Format from "@truffle/codec/format";
 export {
   /**
-   * Here's the decoder output format.
+   * # Codec Output Format
+   *
+   * ## Module information
+   *
+   * This module primarily defines TypeScript types for the output format
+   * used in results provided by packages
+   * `@truffle/decoder@^5.0.0` and `@truffle/codec@^0.1.0`.
+   *
+   * See below for complete listing or continue reading
+   * [Format information](#format-information) to learn about this format.
+   *
+   * ### How to import
+   *
+   * Import either as part of Codec or by itself:
+   *
+   * ```typescript
+   * // when importing entire Codec, use Codec.Format.*:
+   * import * as Codec from "@truffle/codec";
+   *
+   * // or import Format directly:
+   * import { Format } from "@truffle/codec";
+   * ```
+   *
+   * ![Example struct decoding](media://example-struct-decoding.png)
+   *
+   * ## Format information
+   *
+   * This format is intended for use in smart contract and dapp development
+   * tools and libraries, and for use in display contexts, such as when
+   * building on-screen components to show transaction and smart contract
+   * state information.
+   *
+   * This format seeks to provide an exhaustive schema for JavaScript
+   * objects to encode **lossless**, **machine-readable** representations of
+   * all possible Solidity and ABI data types and all possible values of those
+   * types.
+   *
+   * This format targets types and values understood by the
+   * [Solidity programming language](https://solidity.readthedocs.io) and
+   * the [Contract ABI specification](https://solidity.readthedocs.io/en/v0.5.3/abi-spec.html),
+   * within the context of the [Ethereum Virtual Machine](https://ethereum.github.io/yellowpaper/paper.pdf)
+   * (EVM) and in raw data for transactions and logs according to the
+   * [Ethereum JSON RPC](https://github.com/ethereum/wiki/wiki/JSON-RPC).
+   *
+   * Objects in this format may be deeply nested and/or contain circular
+   * dependencies. As such, **do not** serialize objects in this format or
+   * otherwise attempt to display them in full without considering potential
+   * risk. **Objects in this format are for the machine to read, not humans!**
+   * This module provides utilities for inspecting objects in this format,
+   * including the **safe** [[Format.Utils.Inspect.ResultInspector]] wrapper
+   * (for [util.inspect](https://nodejs.org/api/util.html#util_util_inspect_object_options)),
+   * and the **unsafe** [[Format.Utils.Inspect.nativize]] function. For more
```

Comment:
I'm a little wary of mentioning nativize, but if you're going to put it in the docs at all, maybe upfront and marked unsafe is the best way to do so...

Reply:
I originally had it as a snarky remark saying "there's an unsafe alternative which you may find for yourself", but I figured the reader doesn't have enough context at that point to recognize the joke.